integration_tests/src/main/python/orc_test.py

```diff
@@ -486,4 +486,111 @@ def test_orc_read_with_corrupt_files(spark_tmp_path, reader_confs, v1_enabled_li
+
+_aggregate_orc_list_col_partition = ['COUNT']
+_aggregate_orc_list_no_col_partition = ['MAX', 'MIN']
+_aggregate_orc_list = _aggregate_orc_list_col_partition + _aggregate_orc_list_no_col_partition
+_orc_aggregate_pushdown_enabled_conf = {'spark.sql.orc.aggregatePushdown': 'true',
+                                        "spark.sql.sources.useV1SourceList": ""}
+
+def _do_orc_scan_with_agg(spark, path, agg):
+    return spark.read.orc(path).selectExpr('{}(p)'.format(agg))
+
+@pytest.mark.skipif(is_before_spark_330(), reason='Aggregate push down on ORC is a new feature of Spark 330')
+@pytest.mark.parametrize('aggregate', _aggregate_orc_list)
+def test_orc_scan_with_aggregate_pushdown(spark_tmp_path, aggregate):
+    """
+    Spark(330,_) allows aggregate pushdown on ORC by enabling spark.sql.orc.aggregatePushdown.
+    When the spark configuration is enabled we check the following:
+    ---------------------------+
+    | Aggregate | FallBack CPU |
+    +-----------+--------------+
+    |   COUNT   |      Y       |
+    |    MIN    |      Y       |
+    |    MAX    |      Y       |
+    """
+    data_path = spark_tmp_path + '/pushdown.orc'
+    # GPU ORC write with statistics is not correctly working.
+    # Create ORC file in CPU session as a workaround
+    with_cpu_session(lambda spark: spark.range(10).selectExpr("id", "id % 3 as p").write
+                                .mode("overwrite")
+                                .orc(data_path))
+    
+    # fallback to CPU
+    assert_cpu_and_gpu_are_equal_collect_with_capture(
+        lambda spark: _do_orc_scan_with_agg(spark, data_path, aggregate),
+        exist_classes="BatchScanExec",
+        non_exist_classes="GpuBatchScanExec",
+        conf=_orc_aggregate_pushdown_enabled_conf)
+
+@pytest.mark.skipif(is_before_spark_330(), reason='Aggregate push down on ORC is a new feature of Spark 330')
+@pytest.mark.parametrize('aggregate', _aggregate_orc_list_col_partition)
+def test_orc_scan_with_aggregate_pushdown_on_col_partition(spark_tmp_path, aggregate):
+    """
+    Spark(330,_) allows aggregate pushdown on ORC by enabling spark.sql.orc.aggregatePushdown.
+    Note that Min/Max don't push down partition column. Only Count does.
+    This test checks that GPU falls back to CPU when aggregates are pushed down on ORC.
+    When the spark configuration is enabled we check the following:
+    ----------------------------------------------+
+    | Aggregate | Partition Column | FallBack CPU |
+    +-----------+------------------+--------------+
+    |   COUNT   |        Y         |      Y       |
+    """
+    data_path = spark_tmp_path + '/pushdown.orc'
+    # GPU ORC write with statistics is not correctly working.
+    # Create ORC file in CPU session as a workaround
+    # Partition column P
+    with_cpu_session(lambda spark: spark.range(10).selectExpr("id", "id % 3 as p").write
+                                .partitionBy("p")
+                                .mode("overwrite")
```

Comment:
I don't think we want overwrite behavior here, as that implies we're smashing someone else's data (likely another test).  `spark_tmp_path` should be a unique path each time and is automatically cleaned up after the test completes.

Comment applies to the other tests.

Reply:
I may be a little bit confused. `data_path = spark_tmp_path + "file.orc"` I thought the `spark_tmp_path` will be automatically unique for each test.

Reply:
Yes, it should be unique, but in the off chance that somehow something is wrong, we do _not_ want to silently overwrite, correct?